src/core/search/rax_tree.h

```diff
@@ -0,0 +1,149 @@
+#pragma once
+
+#include <absl/types/span.h>
+
+#include <cstdio>
+#include <memory_resource>
+#include <optional>
+#include <string_view>
+
+extern "C" {
+
+#include "redis/rax.h"
+}
+
+namespace dfly::search {
+
+// absl::flat_hash_map/std::unordered_map compatible tree map based on rax tree.
+// Allocates all objects on heap (with custom memory resource) as rax tree operates fully on
+// pointers.
+template <typename V> struct RaxTreeMap {
+  struct FindIterator;
+
+  // Simple seeking iterator
+  struct SeekIterator {
+    friend struct FindIterator;
+
+    SeekIterator() {
```

Comment:
can be `raxStart(&it, NULL)` that sets `RAX_ITER_EOF` flag to the iterator, see below